File: Gotho.BlazorPdf/Ts/PdfMetadata.ts
export class PdfMetadata {
    public author: string;
    public creator: string;
    public keywords: string;
    public producer: string;
    public subject: string;
    public title: string;
    public formatVersion: string;
    public creationDate: Date;
    public modifiedDate: Date;
    public customMetadata: Record<string, string>;

    constructor(author: string, 
                creator: string,
                keywords: string,
                producer: string,
                subject: string,
                title: string,
                formatVersion: string, 
                creationDate: Date, 
                modifiedDate: Date,
                customMetadata: Record<string, string> = {}) {
        this.author = author;
        this.creator = creator;
        this.subject = subject;
        this.keywords = keywords;
        this.producer = producer;
        this.title = title;
        this.formatVersion = formatVersion;
        this.creationDate = creationDate;
        this.modifiedDate = modifiedDate;
        this.customMetadata = customMetadata;
    }
}

File: Gotho.BlazorPdf/Pdf/PdfMetadata.cs
namespace Gotho.BlazorPdf.Pdf;

public class PdfMetadata
{
    public string? Author { get; set; }
    public string? Creator { get; set; }
    public string? Keywords { get; set; }
    public string? Producer { get; set; }
    public string? Subject { get; set; }
    public string? Title { get; set; }
    public string? FormatVersion { get; set; }
    public DateTime? CreationDate { get; set; }
    public DateTime? ModifiedDate { get; set; }
    
    public Dictionary<string, string>? CustomMetadata { get; set; }
}